File: 3429-special-array-i/special-array-i.ts
function isArraySpecial(nums: number[]): boolean {
    let isOddFirst = nums[0] % 2 == 1;
    return nums.every((n, i) => i % 2 == 0 ? isOddFirst ? n % 2 : n % 2 == 0 : isOddFirst ? n % 2 == 0 : n % 2)
};

File: 3429-special-array-i/special-array-i.rs
impl Solution {
    pub fn is_array_special(nums: Vec<i32>) -> bool {
        for i in 0..(nums.len() - 1) {
            if (nums[i] ^ nums[i + 1]) & 1 == 0 {
                return false;
            }
        }
        true
    }
}